### PR TITLE
Update golang and golangci-lint

### PR DESCRIPTION
**Pull request by Diaphteiros**
**How to categorize this PR?**

/area usability
/kind technical-debt
/priority 3

**What this PR does / why we need it**:
Upgrades golang to 1.17 and golangci-lint to v1.46.2

**Special notes for your reviewer**:
The previous golangci-lint version is incompatible with golang 1.18, effectively blocking everybody with that golang version from developing on this project.

**Release note**:

```other developer
Upgrade golang to `1.17`.
```
```other developer
Upgrade golangci-lint to `v1.46.2`.
```

### CODE REVIEW

**Review by achimweigel — APPROVED** [review]
/lgtm

File: pkg/landscaper/installations/subinstallations/helper_test.go [hunk 1]
```diff
@@ -161,7 +161,7 @@ var _ = Describe("SubinstallationsHelper", func() {
 			sortInstallationTemplatesAlphabetically(tmpls)
 			_, err := subinstallations.OrderInstallationTemplates(tmpls)
 			Expect(err).To(HaveOccurred())
-			Expect(err.Error()).To(Equal("Op: EnsureNestedInstallations - Reason: OrderNestedInstallationTemplates - Message: The following cyclic dependencies have been found in the nested installation templates: {c -[e_data]-> e -[d_data]-> d -[c_data]-> c}"))
+			Expect(err.Error()).To(Equal("Op: EnsureNestedInstallations - Reason: OrderNestedInstallationTemplates - Message: The following cyclic dependencies have been found in the nested installation templates: {c -[e_data]-> e -[d_mapping]-> d -[c_target]-> c}"))
```
Comment (reshnm) [inline pkg/landscaper/installations/subinstallations/helper_test.go:164]: @Diaphteiros Are these changes also related to the golang version update?
I'm just curious, whether this slipped in by accident.

Comment (Diaphteiros) [reply pkg/landscaper/installations/subinstallations/helper_test.go:164]: No, this was on purpose ... it has been wrong the whole time, but I just didn't notice until I updated the linter and got linting errors, which, after I fixed them, revealed that the test was wrong.
